lib/nvme/nvme_pcie.c

```diff
@@ -1607,5 +1613,65 @@ nvme_pcie_qpair_process_completions(struct spdk_nvme_qpair *qpair, uint32_t max_
 		nvme_pcie_qpair_complete_pending_admin_request(qpair);
 	}
 
+	if (qpair->ctrlr->state == NVME_CTRLR_STATE_READY) {
+
+		nvme_pcie_qpair_check_timeout(qpair);
+	}
+
 	return num_completions;
 }
+
+void
+nvme_pcie_qpair_check_timeout(struct spdk_nvme_qpair *qpair)
+{
+	uint64_t t02;
+	struct nvme_tracker *tr;
+	struct nvme_request *req;
+	struct nvme_pcie_qpair  *pqpair = nvme_pcie_qpair(qpair);
+
+	if (!TAILQ_EMPTY(&pqpair->outstanding_tr)) {
+
+		/*
+		 * qpair could be either for normal i/o or for admin command. If qpair is admin
+		 * and request is SPDK_NVME_OPC_ASYNC_EVENT_REQUEST, skip to next previous.
+		 */
+
+		tr = TAILQ_LAST(&pqpair->outstanding_tr, nvme_outstanding_tr_head);
```

Comment:
Again, the only time we iterate is for admin_queue and we would have to skip older AERs no matter what.